tooling/minitest/src/pass/lock_issue.rs

```diff
@@ -0,0 +1,70 @@
+use crate::*;
+
+#[test]
+/// What this test wants to check is wether there can be a data race
+/// after a lock handover.
+/// 
+/// let global_0 = 0;
+/// let global_1 = null;
+/// 
+/// fn critical() {
+///     acquire(global_0);
+///     atomic_write(global_1, &global_0);
+///     release(*global_1)
+/// }
+/// 
+/// fn second() {
+///     critical();
+/// }
+/// 
+/// fn main() {
+///     global_0 = create_lock();
+///     let id = spawn(second, null);
```

Comment:
Why not `spawn(critical)`?

Reply:
It was like that because critical should return which was not always legal.
But now it is useful to keep the paths the same length.